lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -93,8 +93,11 @@ def parse(document, options = {})
           attributes['id'] = primary_data['id'] if primary_data['id']
           relationships = primary_data['relationships'] || {}
 
-          filter_fields(attributes, options)
-          filter_fields(relationships, options)
+          attributes = transform_keys(attributes, options)
+          relationships = transform_keys(relationships, options)
```

Comment:
and once that is done, why should there be any change in how `only` and `unless` work.  I mean, transform has already `underscore`d everything..